### ISSUE TITLE
App crashes when enabling

### ISSUE THREAD

**Issue by recallmenot** [MuntashirAkon/BatteryChargeLimiter#24]
On a rooted hotdog (One Plus 7 Pro), when I enable the limit, the app crashes, and from then on every time I want to open it.
Have to reinstall it from fdroid to be able to open it again.
Latest crdroid (android 13).
https://paste.crdroid.net/qWWbKG

**Comment by recallmenot** [comment]
repeated the experiment with the second file from: https://github.com/MuntashirAkon/BatteryChargeLimiter/issues/21
same result: https://paste.crdroid.net/irDvbO

**Comment by Kelvino9** [comment]
Same issue on OnePlus 8T with Android 13 (LineageOS 20)

**Comment by lord-ne** [comment]
I got the same issue when installing from the Play Store, but I didn't get it when compiling from source

**Comment by Retardium** [comment]
I have the same problem, probably related.
Error: 
`java.lang.IllegalArgumentException: io.github.muntashirakon.bcl: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
	at android.app.PendingIntent.checkFlags(PendingIntent.java:401)
	at android.app.PendingIntent.getBroadcastAsUser(PendingIntent.java:671)
	at android.app.PendingIntent.getBroadcast(PendingIntent.java:658)
	at io.github.muntashirakon.bcl.EnableWidget.buildButtonPendingIntent(EnableWidget.kt:2)
	at io.github.muntashirakon.bcl.EnableWidget.updateWidget(EnableWidget.kt:3)
	at io.github.muntashirakon.bcl.activities.MainFragment$$ExternalSyntheticLambda3.onCheckedChanged(Unknown Source:21)
	at android.widget.CompoundButton.setChecked(CompoundButton.java:222)
	at androidx.appcompat.widget.SwitchCompat.setChecked(SwitchCompat.java:1)
	at androidx.appcompat.widget.SwitchCompat.toggle(SwitchCompat.java:1)
	at android.widget.CompoundButton.performClick(CompoundButton.java:144)
	at android.view.View.performClickInternal(View.java:7483)
	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
	at android.view.View$PerformClick.run(View.java:29357)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7884)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
`

**Comment by shuvashish76** [comment]
Also crashing for Android 12.
App version : 1.0.0
<details><summary>Logs</summary>
<p>

```
FATAL EXCEPTION: main
Process: io.github.muntashirakon.bcl, PID: 32235
java.lang.IllegalArgumentException: io.github.muntashirakon.bcl: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
	at android.app.PendingIntent.checkFlags(PendingIntent.java:377)
	at android.app.PendingIntent.getBroadcastAsUser(PendingIntent.java:647)
	at android.app.PendingIntent.getBroadcast(PendingIntent.java:634)
	at io.github.muntashirakon.bcl.EnableWidget.buildButtonPendingIntent(EnableWidget.kt:2)
	at io.github.muntashirakon.bcl.EnableWidget.updateWidget(EnableWidget.kt:3)
	at io.github.muntashirakon.bcl.activities.MainFragment$$ExternalSyntheticLambda3.onCheckedChanged(Unknown Source:21)
	at android.widget.CompoundButton.setChecked(CompoundButton.java:222)
	at androidx.appcompat.widget.SwitchCompat.setChecked(SwitchCompat.java:1)
	at io.github.muntashirakon.bcl.activities.MainFragment.onStart(MainFragment.kt:3)
	at androidx.fragment.app.FragmentStateManager.start(FragmentStateManager.java:8)
	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:11)
	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:73)
	at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:5)
	at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:22)
	at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:1)
	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1457)
	at android.app.Activity.performStart(Activity.java:8259)
	at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3776)
	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2311)
	at android.os.Handler.dispatchMessage(Handler.java:111)
	at android.os.Looper.loopOnce(Looper.java:238)
	at android.os.Looper.loop(Looper.java:357)
	at android.app.ActivityThread.main(ActivityThread.java:8090)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1026)

```
</p>
</details>

**Comment by chientaget** [comment]
it also doesn't work with Xiaomi note 11 MIUI 14 android 13
But with miui 14 android 12 still works

Looking forward to having it fixed

**Comment by MuntashirAkon** [comment]
Sorry, I quite forgot about the issue. I'll publish a new release in the next week if I'm available.

**Comment by ford--prefect** [comment]
Same problem still persisting - whats the status of the new release?

**Comment by MuntashirAkon** [comment]
Unfortunately, I've become too busy to make any open source contribution this month. I'll see if I can do it in the weekend.

**Comment by ford--prefect** [comment]
I don't want to be a pest but the month (and hopefully your stressful time) has past. Any prospect to get a release done? Thanks and don't feel pressed, please.

**Comment by hapklaar** [comment]
Does anyone have a working apk? @lord-ne you say you fixed it by compiling yourself. Could you maybe share your apk?

**Comment by varad-kadam** [comment]
Facing the same issue.
Pixel 7, A14, Stock ROM 

Is there any update?

**Comment by hapklaar** [comment]
> Facing the same issue. Pixel 7, A14, Stock ROM
> 
> Is there any update?

I had great success on my old Samsung tablet with this app: https://github.com/sriharshaarangi/BatteryChargeLimit. Might want to try that.

**Comment by varad-kadam** [comment]
> I had great success on my old Samsung tablet with this app: https://github.com/sriharshaarangi/BatteryChargeLimit. Might want to try that.

I believe this app is forked from the same and ment to be an update.
However, I'll try it out as well, fingers crossed there are no issues.

**Comment by recallmenot** [comment]
for anyone interested:

AccA works for me.

https://github.com/MatteCarra/AccA

https://f-droid.org/en/packages/mattecarra.accapp/

**Comment by MuntashirAkon** [comment]
Fixed in ea9cdc03d9c043d9e8da6dd053203c4bced4ecba

**Comment by theotheroracle** [comment]
yippee !!
